source/deployment/sso-saml-adfs.md

```diff
@@ -0,0 +1,199 @@
+  - Select `Name ID` for the **Incoming claim type**
+  - Select `Unspecified` for the **Incoming name ID format**
+  - Select `E-Mail Address` for the **Outgoing claim type**
+
+Moreover, select the **Pass through all claim values** option. Then click **Finish**.
+
+![adfs_18_configure_incoming_claim.PNG](../../source/images/adfs_18_configure_incoming_claim.PNG)
+
+19) Click **OK** to create the claim rule, and **OK** again to finish creating rules.
+
+20) Open Windows PowerShell as an administrator and run the following command:
+
+`Set-ADFSRelyingPartyTrust -TargetName <display-name> -SamlResponseSignature "MessageAndAssertion"`
+
+where <display-name> is the name you specified in step 4. In our example it would be `mattermost`.
+
+This action will add the signature to SAML messages, making verification successful.
+
+### Export Identity Provider Certificate
+
+Next, we export the identity provider certificate, which will be later uploaded to Mattermost to finish SAML configuration.
+
+21) In ADFS management sidebar, go to **AD FS > Service > Certificates** and double click on the certificate under **Token-signing**
+
+![adfs_19_export_idp_cert_start.PNG](../../source/images/adfs_19_export_idp_cert_start.PNG)
+
+22) In the **Certificate** screen, go to the **Details** tab and click **Copy to File…**, then **OK**. This opens a **Certificate Export Wizard**.
+
+![adfs_20_export_idp_cert_copy.PNG](../../source/images/adfs_20_export_idp_cert_copy.PNG)
+
+23) In the **Certificate Export Wizard** screen, select the option **Base-64 encoded X.509 (.CER)** and click **Next**.
+
+![adfs_21_export_idp_cert_wizard.PNG](../../source/images/adfs_21_export_idp_cert_wizard.PNG)
+
+24) In the **Completing the Certificate Export Wizard** screen, click **Finish**, then **OK**.
+
+You’re now about to finish configuring SAML for Mattermost!
+
+### Configure SAML for Mattermost
+
+25) Start Mattermost server and sign into Mattermost as a System Administrator. Go to **System Console > Authentication > SAML**.
+  - `SAML SSO URL`: **SAML 2.0/W-Federation URL** from the ADFS Endpoints section (see [Basic Requirements](http://docs.mattermost.com/deployment/sso-saml-adfs-basic-requirements.html))
+  - `Identity Provider Issuer URL`: `Relying party trust identifier` from ADFS you specified in step 8.
+  - `Identity Provider Public Certificate`: X.509 Public Certificate you downloaded in step 24.
+
+![adfs_22_mattermost_basics.PNG](../../source/images/adfs_22_mattermost_basics.PNG)
+
+26) (Optional) Configure Mattermost to verify the signature. The `Service Provider Login URL` is the `SAML 2.0 SSO service URL` you specified in ADFS in step 7.
+
+![adfs_23_mattermost_verification.PNG](../../source/images/adfs_23_mattermost_verification.PNG)
+
+27) (Optional) Enable encryption by uploading the Service Provider Private Key and Service Provider Public Certificate if you uploaded the certificate in ADFS in step 6.
```

Comment:
Add a note on pre-installation section to create the keys using the provided script